### DISCUSSION TITLE
Type of cleaning

### DISCUSSION THREAD

**Discussion by ghost** [mrbungle64/ecovacs-deebot.js#347] [Q&A]
How do I determine type type of cleaning that has been performed? I use `this.vacbot.on('CleanLog', async (object) => {}`  to display the results of the last (and previous) cleaning actions, but the returned object doesn't hold the information in the type of cleaning (Quick Scrubbing, Deep Scrubbing or just Vacuuming). The Ecovacs mobile app is able to show it, so there must be a way to determine it. Any hints on how to?

TiA

**Comment by mrbungle64** [comment]
@cactusss 

Yes, there's an attribute `powerMopType` in the payload. I haven't noticed that yet 😉 
I will add it to the output as soon as I find the time to improve the output of the cleaning protocol in general.
